examples/llm/tests/test_moe.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2022 MosaicML Examples authors
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import warnings
+
+import pytest
+import torch
+from composer import Trainer
+from composer.core import Evaluator
+from composer.utils import dist, reproducibility
+from omegaconf import OmegaConf as om
+
+from examples.common.builders import (build_algorithm, build_callback,
+                                      build_dataloader, build_logger,
+                                      build_optimizer, build_scheduler)
+from examples.common.config_utils import log_config
+from examples.llm.src.model_registry import COMPOSER_MODEL_REGISTRY
+
+
+def gpt_tiny_moe_cfg(conf_path='yamls/mosaic_gpt/125m_moe.yaml', pyramid=False):
+    """Create gpt tiny moe cfg."""
```

Comment:
At some point we might want to consider creating a smaller moe to test, esp if we move to github GPU tests (rather than local?)

Reply:
yeah rn the test goes through quite a bit and realistically only tests things when on a multi-GPU, multi-node setup.
(I'm not sure GitHub will enable that; maybe it does, just not sure)